listenbrainz/webserver/static/js/src/listens/ListenCard.tsx

```diff
@@ -91,6 +96,40 @@ export default class ListenCard extends React.Component<
     );
   };
 
+  submitMissingData = () => {
+    const { listen } = this.props;
+    const form = document.createElement("form");
```

Comment:
This seems to me like a very odd way to go about this.

All this should be achievable with the `fetch` API ([docs here](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch)), which will take a javascript object as input (`{name: listen.track_metadata.release_name || "", …}`) 

There's a decent example code with `method: 'POST'` you can use as a base in that page (https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#supplying_request_options)

Reply:
hey, so I wanted to discuss this too, since it indeed is a very odd way to do this, but I assure you, I did try fetch before opting for this obscure method

I had made myself an HTML form that would populate the musicbrainz form, my purpose was to submit it, record the submission in my browser, and copy it as a fetch request so I could use it in my code. But that didn't seem to work out (never redirected). I even tried to manually write the fetch but it **never redirected**. I played around with the redirect field of the object passed to the fetch, and it still didn't work.

I will look into it more, and update you on this. Thank you so much!

Reply:
I recant myself!
As we discussed on IRC, this is actually the right method to achieve the result we are looking for.

The HTML content returned with the request response cannot be opened by the browser when using the fetch API. Booo !

Creating a hidden form and submitting it programmatically is an elegant solution in comparison to pretty much all the other options I thought of, and also less risky (this default browser function works with CORS, cookies, redirects, etc. ; not sure whatever hacky workaround we come up with will…).


I would just suggest to add a `redirect_uri` (as per [the docs](https://musicbrainz.org/doc/Development/Release_Editor_Seeding), bottom of the page) to redirect to LB on successful MB form submission.
I guess `redirect_uri = window.location.href` ?

And also to add an `edit_note` textarea element, something along the lines of  "Seeded by ListenBrainz missing data page: ${window.location.href}"

Reply:
Ah, maybe instead of using redirect_uri you can set the form element's target property to "_blank" , that should open it in a new tab. Haven't tried that myself so take with a pinch of salt :)

Reply:
I also think we should add a comment at the top to explain why we're doing it this way.
Maybe something along the lines of `The POST request returns an html page, which we can't do anything with in fetch. Instead we need to use a native html form, which once submitted will redirect to the MusicBrainz page appropriately.`?

Reply:
Hi, sorry for replying late, I hadn't opened GitHub in the weekend. But I hear you, and I will implement it accordingly. 

>it programmatically is an elegant solution in comparison to pretty much all the other options I thought of, and also less risky

ikr! And I tried to think of potential alternatives, and one idea struck me, that was to create modal and then putting an iframe in there. But I realised that the serious music submitters will go back and forth into checking the MB database so they would need the back and front button of the browser, so the redirection using form seems like the best method till now. Which is why I didn't raise this idea in IRC.


You are also right about adding a comment to describe the reasoning behind the weird solution.

I will also try the `_blank` solution and let you know if it works or not.

> Add an `edit_note` textarea element     

That sounds quite neat, because suppose the user knows that some data is wrong, the person can just fill the form from scratch. (Seems like that was your intention). So I will add that. Thank you so much @MonkeyDo . Once again, really sorry for the late reply :D

Reply:
To be clear, the `edit_note` I was thinking of would just be added to the form to identify that the data was auto-filled by ListenBrainz, the same way other auto-importers do it* and wouldn't be shown to the user on LB.
They can always add to it in the MB form if they wish.

* See for example this edit note: https://musicbrainz.org/edit/84600395#note-84600395-1

Reply:
Ah that's helpful, this makes it seem more legitimate. I am gonna add this. Thanks!